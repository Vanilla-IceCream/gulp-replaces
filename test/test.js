import fs from 'fs';
import es from 'event-stream';
import { File } from 'gulp-util';

import replaces from '../src';

process.chdir(__dirname);

describe('gulp-replaces', () => {
  it('should replace on a stream', (done) => {
    const replacesThisString = {
      'process.env.NODE_ENV': JSON.stringify('development'),
      'process.env.PORT': 3000,
    };

    const file = new File({
      contents: fs.createReadStream('./fixtures/main.js')
    });

    const stream = replaces(replacesThisString);

    stream.on('data', (newFile) => {
      newFile.contents.pipe(es.wait((err, data) => {
        expect(data).toMatchSnapshot();
        done();
      }));
    });

    stream.write(file);
    stream.end();
  });

  it('should replace on a Buffer', (done) => {
    const replacesThisString = {
      'process.env.NODE_ENV': JSON.stringify('development'),
      'process.env.PORT': 3000,
    };

    const file = new File({
      contents: fs.readFileSync('./fixtures/main.js')
    });

    const stream = replaces(replacesThisString);

    stream.on('data', (newFile) => {
      expect(String(newFile.contents)).toMatchSnapshot();
      done();
    });

    stream.write(file);
    stream.end();
  });
});
