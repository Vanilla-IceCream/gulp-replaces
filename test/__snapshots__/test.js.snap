// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gulp-replaces should replace on a Buffer 1`] = `
"const env = {
  NODE_ENV: \\"development\\",
  PORT: 3000,
};
"
`;

exports[`gulp-replaces should replace on a stream 1`] = `
Object {
  "data": Array [
    99,
    111,
    110,
    115,
    116,
    32,
    101,
    110,
    118,
    32,
    61,
    32,
    123,
    10,
    32,
    32,
    78,
    79,
    68,
    69,
    95,
    69,
    78,
    86,
    58,
    32,
    34,
    100,
    101,
    118,
    101,
    108,
    111,
    112,
    109,
    101,
    110,
    116,
    34,
    44,
    10,
    32,
    32,
    80,
    79,
    82,
    84,
    58,
    32,
    51,
    48,
    48,
    48,
    44,
    10,
    125,
    59,
    10,
  ],
  "type": "Buffer",
}
`;
